### PR TITLE
Allow authorization request resolver to be changed for the OAuth2 client configuration

### PR DESCRIPTION
**Pull request by spoptchev**
Fixes gh-12430. 

See issue for details.

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java [hunk 1]
```diff
@@ -3831,9 +3831,30 @@ public final class OAuth2ClientSpec {
...
+
 		private OAuth2ClientSpec() {
 		}
 
+		/**
+		 * Sets the resolver used for resolving {@link OAuth2AuthorizationRequest}'s.
+		 * @param authorizationRequestResolver the resolver used for resolving
+		 * {@link OAuth2AuthorizationRequest}'s
+		 * @return the {@link OAuth2ClientSpec} for further configuration
+		 */
```
Comment (jgrandja) [inline config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java:3844]: Add `@since 6.1`

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @spoptchev !

Can you also please add a test similar to `OAuth2LoginTests.oauth2LoginWhenCustomObjectsThenUsed()`. We should test if a custom `authorizationRequestResolver` is set then it is used. I think all you need to modify is `OAuth2ClientSpecTests.oauth2ClientWhenCustomObjectsThenUsed()`.

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java (hunk 1 above)
Comment (spoptchev) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java:3845]: Done 👍

**Comment by spoptchev** [comment]
@jgrandja I added the test. Let me know if something is missing.

**Comment by jgrandja** [comment]
Thanks for the updates @spoptchev. This is now merged.

FYI, I added a polish commit with some minor updates in order to get this merged as I am off until the new year.

Enjoy the holidays!

**Comment by spoptchev** [comment]
@jgrandja thank you for the integration.

Happy holidays 🎄

**Pull request closed without merging**